packages/xsnap/test/xsnap-eof.test.js

```diff
@@ -0,0 +1,181 @@
+import test from 'ava';
+import * as proc from 'child_process';
+import * as os from 'os';
+import fs from 'fs';
+import { fileURLToPath, URL } from 'url';
+import { tmpName } from 'tmp';
+import { text } from 'stream/consumers';
+import { makeNetstringWriter } from '@endo/netstring';
+import { makeNodeWriter } from '@endo/stream-node';
+import { makePromiseKit } from '@endo/promise-kit';
+import { options } from './message-tools.js';
+import { xsnap, QUERY_RESPONSE_BUF } from '../src/xsnap.js';
+
+test('xsnap-worker complains while waiting for answer when parent is killed', async t => {
+  const exitedPKit = makePromiseKit();
+
+  const p =
+    /** @type {import("child_process").ChildProcessWithoutNullStreams} */ (
+      proc.fork(
+        fileURLToPath(new URL('./fixture-xsnap-eof.js', import.meta.url)),
+        { stdio: ['ignore', 'pipe', 'pipe', 'ipc'] },
+      )
+    );
+
+  const outP = text(p.stdout);
+  const errP = text(p.stderr);
+
+  p.on('error', err => {
+    exitedPKit.reject(err);
+  });
+
+  p.on('exit', (code, signal) => {
+    exitedPKit.resolve({ code, signal });
+  });
+
+  const [strOut, strErr, { code: nodeExitCode, signal: nodeExitSignal }] =
+    await Promise.all([outP, errP, exitedPKit.promise]);
+
+  t.is(strOut, '', 'stdout must be empty');
+  t.regex(
+    strErr,
+    /Has parent died\?/,
+    'stderr must contain "Has parent died?"',
+  );
+  t.is(nodeExitCode, null, 'exit code must be null');
+  t.is(nodeExitSignal, 'SIGKILL', 'exit signal must be "SIGKILL"');
+});
+
+async function spawnReflectiveWorker(handleCommand) {
+  const exitedPKit = makePromiseKit();
+  let xsnapProcess;
+
+  /** @type {typeof import('child_process').spawn} */
```

Comment:
I think this will sort out the typing complaints:
```js
  /**
   * @type {import('child_process').ChildProcessWithoutNullStreams & {stdio: [
   *   import('stream').Writable,
   *   import('stream').Readable,
   *   import('stream').Readable,
   *   import('stream').Duplex,
   *   import('stream').Duplex,
   * ]}}
   */
  let xsnapProcess;

  const spawnSpy = (...args) => {
    // @ts-expect-error untyped spread and type narrowing
    xsnapProcess = proc.spawn(...args);
    return xsnapProcess;
  };

  // @ts-expect-error overloaded signature for spawn
  const ioOptions = options({ spawn: spawnSpy, os: os.type(), fs, tmpName });
  const vat = await xsnap({ ...ioOptions, handleCommand, stderr: 'pipe' });
  await vat.evaluate(
    'function handleCommand(message) { issueCommand(new Uint8Array().buffer); };',
  );

  // @ts-expect-error use before assignment
  assert(xsnapProcess !== undefined);

  const toXsnap = xsnapProcess.stdio[3];
  const fromXsnap = xsnapProcess.stdio[4];
  …
```

Reply:
Heh, looks like @mhofman already did pretty much this.

Reply:
Yeah I had to fight TS for a while!